algo/src/main/java/org/neo4j/graphalgo/impl/metaPathComputation/ComputeAllMetaPathsSchemaFull.java

```diff
@@ -1,212 +1,99 @@
+    public void computeMetaPathFromNodeLabel(int nodeID, int metaPathLength) {
         ArrayList<Integer> initialMetaPath = new ArrayList<>();
-        initialMetaPath.add(nodeID); //because node is already type (of nodes in the real graph)
+        initialMetaPath.add(reversedLabelDictionary.get(nodeID)); //because nodeID is already a type of nodes in the real graph//convert to heavyGraph nodeType
+
+        addAndLogMetaPath(initialMetaPath);
         computeMetaPathFromNodeLabel(initialMetaPath, nodeID, metaPathLength - 1);
     }
 
-    private void computeMetaPathFromNodeLabel(ArrayList<Integer> pCurrentMetaPath, int pCurrentInstance, int pMetaPathLength) {
-        Stack<ArrayList<Integer>> st_allMetaPaths = new Stack();
-        Stack<Integer> st_currentNode = new Stack();
-        Stack<Integer> st_metaPathLength = new Stack();
-        st_allMetaPaths.push(pCurrentMetaPath);
-        st_currentNode.push(pCurrentInstance);
-        st_metaPathLength.push(pMetaPathLength);
-
-        ArrayList<Integer> currentMetaPath;
-        int currentInstance;
-        int metaPathLength;
-
-        while(!st_allMetaPaths.empty() && !st_currentNode.empty() && !st_metaPathLength.empty())
-        {
-            currentMetaPath = st_allMetaPaths.pop();
-            currentInstance = st_currentNode.pop();
-            metaPathLength = st_metaPathLength.pop();
-
-            if (metaPathLength <= 0) {
-                continue;
-            }
-
-            HashSet<AbstractMap.SimpleEntry<Integer, Integer>> outgoingEdges = new HashSet<>();
-            outgoingEdges.addAll(adjacentNodesDict.get(currentInstance));
-            for (AbstractMap.SimpleEntry<Integer, Integer> edge : outgoingEdges) {
-                ArrayList<Integer> newMetaPath = copyMetaPath(currentMetaPath);
-                int nodeID = edge.getKey();
-                int edgeID = edge.getValue();
-                newMetaPath.add(edgeID);
-                newMetaPath.add(nodeID);
-
-                synchronized (duplicateFreeMetaPaths) {
-                    // add new meta-path to threads
-                    String joinedMetaPath;
-                    joinedMetaPath = newMetaPath.stream().map(Object::toString).collect(Collectors.joining("|"));
-                    duplicateFreeMetaPaths.add(joinedMetaPath);
-                    out.println(joinedMetaPath);
-                }
-
-                st_allMetaPaths.push(newMetaPath);
-                st_currentNode.push(nodeID);
-                st_metaPathLength.push(metaPathLength - 1);
-                //debugOut.println("finished recursion of length: " + (metaPathLength - 1));
-            }
+    private void computeMetaPathFromNodeLabel(ArrayList<Integer> currentMetaPath, int currentInstance, int metaPathLength) {
+        if (metaPathLength <= 0) return;
+
+        HashSet<Pair> neighbourNodes_edges = schema.get(currentInstance);
```

Comment:
Java convention: camel case

Reply:
the name neighbourNodesEdges would be somewhat misleading

Reply:
Then we have to choose an other name. We cannot use camel case or Java will explode.